core/vm/interpreter.go

```diff
@@ -228,7 +234,7 @@ func (in *Interpreter) Run(snapshot int, contract *Contract, input []byte) (ret
 		// if the operation returned a value make sure that is also set
 		// the last return data.
 		if res != nil {
```

Comment:
I'm not sure this is sufficient. 

In the case of `CALL`, the `res` may be `nil` for various reasons, e.g. the balance being to low: 
```
	if !evm.Context.CanTransfer(evm.StateDB, caller.Address(), value) {
		return nil, gas, ErrInsufficientBalance
	}
```
The way it's written, I don't think `RETURNDATA` would be cleared. I believe the returndata should be cleared immediately when `CALL` is invoked. 

Further, I believe that also `CREATE` would clear the `RETURNDATA`. But I'm not 100% certain in this case. @pirapira what's your take ?